File: SecondYear/Parth27904_ParthAgarwal_2226CSE(AI&ML)1015_2/Week_2/reverse_bits.java
public class reverse_bits {
	public static long reverseBits(long n) {
        long reversedN = 0;
        int bits = 32;
        for (int i = 0; i < bits; i++) {
            reversedN <<= 1;
            reversedN |= (n & 1);
            n >>= 1;
        }
        return reversedN;
	}
}

File: SecondYear/Junior-Hawk_ArchitAgarwal_2226cseai1186_2/Week_2/reverse_bits.cpp
long reverseBits(long n) {
    vector<int> bits(32,0);
    int i =0,c=0;
    while(n>0){
        int rem = n%2;
        if(rem==1) c++;
        bits[i] = rem;
        i++;
        n/=2;
    }
    long k=0;
    for(int i=0;i<32;i++){
        if(bits[i]==1){
            k += pow(2,32-i-1);
            c--;
        }
        if(c==0) break;
    }
    return k;
}